api-library/api.yaml

```diff
@@ -186,6 +188,183 @@ paths:
+    put: 
+      tags: 
+        - Users
+
+      # parameters: 
+      #   - name: auth
+      #     in: header
+      #     description: an authorization header
+      #     required: true
+      #     schema:
+      #       type: string
+      
+      requestBody:
+        content:
+          application/json: 
+            schema:
+              $ref: '#/components/schemas/User'
+      summary: update user's data
+      responses:
+        "200": 
+          description: successful update 
+          content:
+            application/json: 
+              schema:
+                $ref: '#/components/schemas/User'
+        
+        "422": 
+          description: validation error
+          content:
+            application/json: 
+              schema:
+                $ref: '#/components/schemas/User'
+        "500":
+          $ref: "#/components/responses/InternalServerError"
+    delete:
+      tags: 
+        - Users
+      summary: delete's user data 
+      responses:
+        "200": 
+          $ref: '#/components/responses/ok'
+        "401": 
+          description: user is unidentified 
+          content: 
+            application/json: 
+              schema:
+                $ref: '#/components/schemas/Unauthorized'
+  /user/verify/{code}: 
+    post:
+      tags: 
+        - Users
+        - Auth
+      summary: verifiying the code sent to the user's email 
+      parameters: 
+        - name: code
+          in: path 
+          description: the code in the email
+          required: true
+          schema:
+            type: string 
+            
+      responses:
+        "200": 
+          $ref: '#/components/responses/ok'
+        
+        "401": 
+          $ref: '#/components/responses/Unauthorized'
+  /user/{id}: 
+    get: 
+      tags: 
+        - Users
+        
+      summary: get user's info by id
+      parameters: 
+        - name: id
+          in: path 
+          description: user's id 
+          required: true
+          schema:
+            type: string 
+            
+      responses:
+        "200":
+          description: user's info
+          content:
+            application/json: 
+              schema:
+                $ref: '#/components/schemas/UsersForUsers'
+                
+        "404": 
+          $ref: '#/components/responses/NotFound'
+        
+        "401": 
+          $ref: '#/components/responses/Unauthorized'
+  /login: 
+    post: 
+      tags: 
+        - Users
+        - Auth
+      summary: user's login 
+      requestBody:
+        content:
+          application/json: 
+            schema:
+              $ref: '#/components/schemas/UserAuth'
+      
+      responses:
+        "200": 
+          $ref: '#/components/responses/ok'
+        
+        "422": 
+          description: user's loggin credentioals didn't pass the validation step 
+          content:
+            application/json: 
+              schema:
+                $ref: '#/components/schemas/UserAuth'
```

Comment:
I think this could also be generalized in the responses section.

Reply:
should it? it's only being mentioned twice.